### ISSUE TITLE
nested tsconfig.json and paths

### ISSUE THREAD

**Issue by plandem** [TypeStrong/ts-loader#580]
Hi

sources like this:
```js
/tsconfig.base.json
/src/web/...
/src/web/tsconfig.json
/src/shared/...
/src/desktop/...
/src/desktop/tsconfig.json
```
I have next tsconfig:

for /web
```json
{
  "compilerOptions": {
    "jsx": "react",
    "paths": {
      "app/*": ["./web/*"],
      "shared/*": ["./shared/*"]
    }
  },
  "extends": "../../tsconfig.base"
}
```

for /desktop
```json
{
  "compilerOptions": {
    "paths": {
      "app/*": ["./desktop/*"],
      "shared/*": ["./shared/*"]
    }
  },
  "extends": "../../tsconfig.base"
}
```

and base
```json
{
  "compilerOptions": {
    "target": "es6",
    "module": "es6",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "allowJs": true,
    "strictNullChecks": true,
    "noImplicitAny": false,
    "allowSyntheticDefaultImports": true,
    "baseUrl": "src"
  },
  "exclude": [
    "node_modules"
  ]
}
```

tsc works fine, but I can't make ts-loader to work properly.
I have 2 applications and both use **app**  alias. So I can't use one webpack.config for ts-loader, because I can't set alias property there.

here is webpack:
```js
const webpack = require('webpack');
const path = require('path');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;

const isProduction = process.env.NODE_ENV === 'production';
const visualize = false;

const config = {
	devtool: false,
	target: 'node',
	entry: {
		web: './src/web/index.tsx',
		desktop: './src/desktop/index.ts',
		common: [
			'redux',
			'node-polyglot',
		],
		core: [
			'react',
			'react-dom',
			'moment',
			'react-redux',
		],
	},

	output: {
		filename: `[name].js`,
		chunkFilename: '[name]-chunk.js',
		path: path.resolve(__dirname, 'dist/bundle'),
		publicPath: '/'
	},

	devServer: {
		hot: true,
		contentBase: path.resolve(__dirname, 'dist/bundle'),
		publicPath: '/',
		port: 8001,
		stats: { modules:false }
	},
	externals: [
		'electron',
	],
	resolve: {
		extensions: ['.ts', '.tsx', '.js', '.jsx'],
		alias: {
			//can't be set :(
		},
		plugins: [
			//custom plugins doesn't work with ts-loader
			new webpack.NormalModuleReplacementPlugin(new RegExp('app/'), 'xxxx')
		]
	},
	module: {
		rules: [
			{ test: /\.tsx?$/, exclude: /node_modules/, use: [{ loader: 'babel-loader', options: {
				presets: ['electron'],
				plugins: ['lodash', ['import', {
					libraryName: 'antd',
					style: true,
				}]],
				babelrc: false
			}}, 'ts-loader']},
			{ test: /\.html?$/, use: { loader: 'file-loader', query: { name: '[name].[ext]'}}},
			{ test: /\.less$/i, use: ExtractTextPlugin.extract([ 'css-loader', 'less-loader' ])},
		]
	},
	plugins: [
		(visualize ? new BundleAnalyzerPlugin({
			analyzerMode: 'static'
		}) : null),

		new ExtractTextPlugin(`${type}.css`),

		new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),

		new webpack.optimize.CommonsChunkPlugin({
			names: ['web', 'core', 'common'], minChunks: Infinity,
		}),

		(!isProduction ? new webpack.HotModuleReplacementPlugin() : null),
	].filter(p => p),
};

module.exports = config;
```

how to solve it? 

P.S.: I would like to use one webpack.config

**Comment by johnnyreilly** [comment]
It sounds like each entry represents a different TypeScript application. You probably need to make use of the ts-loader `instance` option which facilitates this. See example usage here: https://github.com/TypeStrong/ts-loader/blob/master/test/comparison-tests/instance/webpack.config.js

**Comment by plandem** [comment]
@johnnyreilly 

ok, as I understand about **instance**:
```
	{ test: /\.tsx?$/, exclude: /node_modules/, use: [babelLoader, { loader: 'ts-loader' }]},
	{ test: /src\/web\/(.*)\.tsx?$/, exclude: /node_modules/, use: [babelLoader, { loader: 'ts-loader', options: { instance: 'web'} }]},
```

still doesn't work:
```
ERROR in ./src/desktop/index.ts
Module not found: Error: Can't resolve 'app/main' in '/Users/andrey/Projects/desktop/catapult/src/desktop'
 @ ./src/desktop/index.ts 3:12-31
```

**Comment by plandem** [comment]
Actually, I'm not sure that error come from TS - errors from tsc has prefixes TSxxxx.

**Comment by Gillardo** [comment]
i am trying to do exactly the same, did you solve this??

**Comment by lddubeau** [comment]
@plandem Did you try replacing `baseUrl` with an absolute path? I ran into an issue that feels similar to what you are reporting here. To see what would happen, I replaced `baseUrl` with an absolute path, and the module resolution issue disappeared.

Using an absolute path is not a solution or even a viable workaround but it suggests that `ts-loader` is probably not computing paths correctly under some usage scenarios.
